dts/bindings/serial/uart.yaml

```diff
@@ -24,6 +24,11 @@ properties:
       category: required
       description: Initial baud rate setting for UART
       generation: define
+    hardware-fc-rtscts:
+      type: boolean
+      category: optional
+      description: Initial hardware flow control setting for UART
```

Comment:
IIRC, behavior of optional booleans is as follows:
If available in the node, generated define will be set to true
If not available, generated define will be set to true
So, I'll put the description as follows: "Hardware flow control enabled at boot time".